### PR TITLE
[MB-2105] Add list all to get state of the world for matching attributes

### PR DESCRIPTION
**Pull request by katrogan**

### CODE REVIEW

File: protos/flyteidl/admin/matchable_resource.proto [hunk 1]
```diff
@@ -64,3 +64,23 @@ message MatchingAttributes {
     ExecutionClusterLabel execution_cluster_label = 4;
   }
 }
+
+message MatchableResourceConfiguration {
```
Comment (EngHabu) [inline protos/flyteidl/admin/matchable_resource.proto]: nit: mind adding docs for the messages?

File: protos/flyteidl/admin/matchable_resource.proto [hunk 2]
```diff
@@ -64,3 +64,23 @@ message MatchingAttributes {
...
+
+    string domain = 2;
+
+    string project = 3;
+
+    string workflow = 4;
+}
+
+message ListMatchableResourcesRequest {
+    string domain = 1;
```
Comment (EngHabu) [inline protos/flyteidl/admin/matchable_resource.proto]: Why only domain?

Comment (katrogan) [reply protos/flyteidl/admin/matchable_resource.proto]: we scope everything in the system by domain? but i can remove if you think it's not a great requirement!

File: protos/flyteidl/admin/matchable_resource.proto [hunk 3]
```diff
@@ -64,3 +64,25 @@ message MatchingAttributes {
...
+// Represents a custom set of attributes applied for either a domain; a domain and project; or
+// domain, project and workflow name.
+message MatchableResourceConfiguration {
+    MatchingAttributes attributes = 1;
+
+    string domain = 2;
+
+    string project = 3;
+
+    string workflow = 4;
```
Comment (anandswaminathan) [inline protos/flyteidl/admin/matchable_resource.proto]: You would need Priority and Launch plan as well

File: protos/flyteidl/admin/matchable_resource.proto [hunk 4]
```diff
@@ -64,3 +64,25 @@ message MatchingAttributes {
...
+}
+
+// Request all matching resource attributes.
+message ListMatchableResourcesRequest {
```
Comment (anandswaminathan) [inline protos/flyteidl/admin/matchable_resource.proto]: Do you want to name this matchable ?

File: protos/flyteidl/admin/matchable_resource.proto (hunk 3 above)
Comment (katrogan) [reply protos/flyteidl/admin/matchable_resource.proto]: +1 for launch plan, thanks for the catch
as for priority, i'd rather the server render the resolved state of the world so the caller of this endpoint can see what the actual applied values are

File: protos/flyteidl/admin/matchable_resource.proto (hunk 4 above)
Comment (katrogan) [reply protos/flyteidl/admin/matchable_resource.proto]: sure, done

**Comment by katrogan** [comment]
ping @anandswaminathan

**Comment by anandswaminathan** [comment]
+1

**Pull request merged**